Update readme and release procedure after 0.2.0 release (#1300)
C2SM/icon4py: README.md

## Project Structure

> The repository is organized as a _monorepo_, where various ICON model components and utilities are developed as independent Python namespace packages in subfolders. The `icon4py` package at the root collects specific versions of the different components as dependencies and is published on PyPI as a meta-package. Individual namespace packages are also available on PyPI and can be installed independently.

Reviewer: ```suggestion
The repository is organized as a _monorepo_, where various ICON model components and utilities are developed as independent Python namespace packages in subfolders. The `icon4py` root package collects specific versions of the different components and stores them as dependencies. `icon4py` is published on PyPI as a meta-package. Individual namespace packages are also available on PyPI and can be installed independently.
```

Author: Is this also ok @nfarabullini:
```suggestion
The repository is organized as a _monorepo_, where various ICON model components and utilities are developed as independent Python namespace packages in subfolders. The `icon4py` root package collects specific versions of the different components as dependencies. `icon4py` is published on PyPI as a meta-package. Individual namespace packages are also available on PyPI and can be installed independently.
```
? I removed the "stored as dependencies" part, since they're not really "stored".

Reviewer: yes!

## License

ICON4Py is licensed under the terms of the BSD-3-Clause.

## Installation Instructions

All ICON4Py packages can be installed through PyPI with the [`icon4py` meta package](https://pypi.org/project/icon4py/).

> We recommend using [`uv`](https://docs.astral.sh/uv/) for both regular installation and development workflows. See the [installation instructions](https://docs.astral.sh/uv/getting-started/installation/)) for details on installing `uv`.

Reviewer 2: ```suggestion
We recommend using [`uv`](https://docs.astral.sh/uv/) for both regular installation and development workflows. See the [installation instructions](https://docs.astral.sh/uv/getting-started/installation/) for details on installing `uv`.
```

```bash
uv pip install --prerelease=allow icon4py
```

**Note:** `--prerelease=allow` is currently required for the `dace` dependency, which is pinned to a prerelease version.

### Extras

> The `icon4py` meta-package provides the following optional extras. The same extras apply in development workflows by using `uv sync --extra <name>`.

Reviewer 2: There's nothing wrong here, just an optional suggestion in case some user doesn't know much about `uv`:
```suggestion
The `icon4py` meta-package provides the following optional extras. The same extras apply in development workflows by using `uv sync --extra <name1> --extra <name2> ...`.
```

| Extra         | Description                                                                                                                                            |
| ------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------ |
| `all`         | All extras listed below, except the GPU extras.                                                                                                        |
| `cuda12`      | Required for CUDA support with CUDA version 12. Requires the CUDA toolkit to be installed.                                                             |
| `cuda13`      | Same as `cuda12` for CUDA version 13.                                                                                                                  |
| `rocm7`       | Same as `cuda*` for ROCm on AMD GPUs.                                                                                                                  |

> | `distributed` | Enables Support for distributed multi-node runs. Requires an MPI implementation and Boost headers. See below for fixes to installation issues on Alps. |

Reviewer: ```suggestion
| `distributed` | Enables support for distributed multi-node runs. Requires an MPI implementation and Boost headers. See below for fixes to installation issues on Alps. |
```

> | `fortran`     | Enables `icon4py-bindings` as a dependency, which provides an interface for using ICON4Py in ICON Fortran. Requires C and Fortran compilers.           |

Reviewer: ```suggestion
| `fortran`     | Enables `icon4py-bindings` as a dependency, which provides an interface between ICON4Py and ICON Fortran. Requires C and Fortran compilers.           |
```
